src/classes/UTIL_OrgTelemetry_SObject_BATCH.cls

```diff
@@ -0,0 +1,119 @@
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2018
+* @group Utilities
+* @description To prevent a timeout error counting Opportunity records using specific filters, this batch job
+* is built to be generic enough to be executed in different configurations to count whatever large object requires
+* counting. Stateful is used to hold the count variables and an Enum is used to define the mode the batch job
+* is executing in.
+*/
+public class UTIL_OrgTelemetry_SObject_BATCH implements Database.Batchable<SObject>, Database.Stateful {
+
+    /**
+     * @description BatchJobMode is to add support for other query types in this same batch job
+     * that aren't yet needed, but may be in the future
+     */
+    public enum TelemetryBatchJobMode {
+        RECURRING_DONATIONS
+    }
+
+    /** @description The job mode this particular batch iteration is executing in */
+    private UTIL_OrgTelemetry_SObject_BATCH.TelemetryBatchJobMode jobMode;
+
+    /** @description Stateful counter vars used for the different counting jobs */
+    private Integer countAllRDOpps = 0;
+    private Integer countOpenRDOpps = 0;
+
+    private static final String recurringDonationOpenStatusValue = System.Label.npe03.RecurringDonationOpenStatus;
+
+
+    /**
+     * @description Constructor
+     */
+    public UTIL_OrgTelemetry_SObject_BATCH(UTIL_OrgTelemetry_SObject_BATCH.TelemetryBatchJobMode jobMode) {
+        this.jobMode = jobMode;
+    }
+
+    /**
+     * @description Execute the appropriate query to use in this batch job.
+     */
+    public Database.QueryLocator start(Database.BatchableContext context) {
+        switch on (jobMode) {
+            when RECURRING_DONATIONS {
+                return Database.getQueryLocator(
+                        [SELECT Id, npe03__Recurring_Donation__r.npe03__Open_Ended_Status__c
+                        FROM Opportunity
+                        WHERE npe03__Recurring_Donation__c != null]);
+            }
+        }
+        return null;
```

Comment:
The `start()` method cannot return `null` for the QueryLocator. For example, when the batch is executed with the `jobMode` being null
```
Database.executeBatch(new UTIL_OrgTelemetry_SObject_BATCH(null), 2000);
```
then the following error is issued:
> First error: Start did not return a valid iterable object.

I guess this is fine since the batch is not global, but wanted to notify you.